### PR TITLE
Add json-schema check to gameStateUtil.save

### PR DESCRIPTION
**Pull request by rachidlamouri**

### CODE REVIEW

File: src/utils/gameStateUtil.ts [hunk 1]
```diff
@@ -25,7 +26,14 @@ export const gameStateUtil = {
 
     return gameState;
   },
-  save: (state: GameState): void => {
-    databaseUtil.save(state);
+  save: (state: object): void => {
```
Comment (rachidlamouri) [inline src/utils/gameStateUtil.ts:29]: I think I like `unknown` better.

File: src/utils/gameStateUtil.ts [hunk 2]
```diff
@@ -25,7 +26,14 @@ export const gameStateUtil = {
...
+    if (isGameState(state)) {
+      databaseUtil.save(state);
+      return;
+    }
+
+    throw new DeveloperError(
+      'Attempted to save an invalid game state. Did you forget to compile the game state schema?',
```
Comment (jshemas) [inline src/utils/gameStateUtil.ts:36]: Should we add the `state` they are trying to save here?

Comment (rachidlamouri) [reply src/utils/gameStateUtil.ts:36]: I can't decide if I want to console.log it or stringify it and write to a file. So i'm going to defer to whoever gets this error first.

File: src/utils/gameStateUtil.ts (hunk 1 above)
Comment (rachidlamouri) [reply src/utils/gameStateUtil.ts]: Updated

File: src/utils/gameStateUtil.ts (hunk 2 above)
Comment (jshemas) [reply src/utils/gameStateUtil.ts:36]: It will be a race.

**Review by jshemas — APPROVED** [review]

File: src/utils/gameStateUtil.ts (hunk 2 above)
Comment (rachidlamouri) [reply src/utils/gameStateUtil.ts:36]: 🐀

**Pull request merged**